sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidRexExecutor.java

```diff
@@ -91,7 +94,10 @@ public void reduce(
           // as a primitive long/float/double.
           // ExprEval.isNumericNull checks whether the parsed primitive value is null or not.
           if (!constExp.getType().isNullable() && exprResult.isNumericNull()) {
-            throw new UnsupportedSQLQueryException("Illegal DATE constant: %s", constExp);
+            throw DruidException.user("Illegal DATE constant")
```

Comment:
Here again meaningful values (the date) that should get back to the user are potentially going to be ellided because they are in the context.